lib/worker.js

```diff
@@ -274,34 +275,41 @@ class AssetComputeWorker {
             }
             return;
         }
-
+        // batch post-process is all or none. so using currentPostProcessing(is timer) as postProcessing(is duration)
+        this.timers.currentPostProcessing = new Timer();
+        let ppRendition = 0;
         for (let rendition of this.renditions) {
-            // start post-processing timer 
-            // batch post-process is all or none. i.e currentCallbackProcessing = callbackProcessing
-            // we dont need this.timers.currentPostProcessing metrics for batch
-            // if we do it should be similar => currentPostProcessing = postProcessing
-            const postProcessingTimer = this.timers.currentPostProcessing = new Timer();
+            // start post-processing timer - used for logging current interation
+            const postProcessingTimer = new Timer();
             const shouldDoPostProcess = await this.shouldPostProcess(rendition);
             if (shouldDoPostProcess) {
                 try {
                     rendition = await this.doPostProcess(rendition);
                     console.log(`Batch Postprocessing finished in ${postProcessingTimer.durationSec()} seconds: ${rendition.name}`);
+                    ppRendition++;
                 } catch (err) {
-                    console.error(`rendition post-processing for batch failed with error after ${postProcessingTimer.durationSec()}: ${err.message || err}`);
+                    console.error(`rendition post-processing for ${rendition.name} failed with error after ${postProcessingTimer.durationSec()}: ${err.message || err}`);
+                    console.error(`rendition post-processing for batch failed after ${this.timers.currentPostProcessing.durationSec()}`);
                     await this.handleBatchPostProcessFailure(err);
                     return;
                 }
             } else {
-                const err = new GenericError('batch worker renditions postprocess applies only for all renditions');
+                console.log(`No postprocessing done for rendition ${rendition.name}`);
+            }
+            // by convention if a even a single rendition is not postprocess eligible then failAll.
+            // following block is only for reqeust with multiple rendition and if few are postprocess ineligible
+            // skip: if only 1 rendition and is not post process eligible
+            // fail fast: if more than one rendition and one or more failed
+            if(ppRendition > 0 && !shouldDoPostProcess) {
```

Comment:
I am still confused about this. this is in case there has been at least one post-processed asset and another asset in the batch is not supposed to be post-processed.

1. we then can make `ppRendition` into a boolean instead of a counter. we only need to track whether at least 1 rendition was post processed. in this case lets make it a boolean and call it `postProcessed`

2. in the case where one of the renditions in the batch is post-processed and another is not post-processed, I would argue we don't need to fail here. we should fail the entire batch if one of the renditions fails during post-processing. but if one rendition needs post-processing and others don't I do not think we should fail the entire renditions.

Reply:
I think the logic is wrong, here. A rendition that does not need post-processing will also have shouldDoPostProcess return false (remember, post-processing needs a flag set). As far as I understand the describing comment, a rendition not requesting post-processing would also make the batch fail here... Which is not what we want, as some renditions may indeed not need post-processing.

Reply:
(Remember that the post-processing requesting flag is per rendition, not per batch).

Reply:
@jdelbick @tmathern 
Implementation to change after discussion on SLACK: https://cq-dev.slack.com/archives/GAMJLHSBH/p1597871482310900?thread_ts=1597870229.284100&cid=GAMJLHSBH

@jdelbick I remember trying the Boolean logic which I think failed existing tests.